### **Username: suneelraju160**
### **Name: Kucharlapati Venkata Suneel Raju**
### **Data Science latest assignments - 16 Aug 2025**
### **Assignment 6: Confidence Interval and Hypothesis Testing**
### **Date: 08 Dec 2025**


Hypothesis testinh H0 claim:

### **Restaurant owners's claim that there are increase in weekly operational costs.**

In [7]:
from math import sqrt, erf

# Given data

# sample mean weekly cost 'x_bar' (Rs.)
x_bar = 3050.0

# sample size 'n'
n = 25

# mean units produced (X)
mu_x = 600.0

# sd of units produced (X)
SD_x = 25.0

# Theoretical mean of weekly cost according to model
mu_theoretical = 1000 + 5 * mu_x

# Population sd of weekly cost
SD_w = 5 * SD_x

# Standard error of the sample mean
se = SD_w / sqrt(n)

# z vlue by test statistic
z_stat = (x_bar - mu_theoretical) / se

# Standard normal CDF using error function
def std_norm_cdf(z):
    return 0.5 * (1 + erf(z / sqrt(2)))

# For one-sided Ha: mu > mu_theoretical, p-value = P(Z >= z_stat) = 1 - Phi(z_stat)
p_value_one_sided = 1 - std_norm_cdf(z_stat)

# Significance level and critical z (one-sided)
alpha = 0.05

# approximate critical z for 95th percentile of standard normal:
z_crit_one_sided = 1.6448536269514722


print("Hypothesis Test: Are weekly costs higher than the theoretical model?")
print(f"Sample mean (x̄): ", x_bar)
print(f"Theoretical mean (μ): ", mu_theoretical)
print(f"Population sd of weekly cost: ", SD_w)
print(f"Sample size (n): ", n)
print(f"Standard error : ", se)
print(f"Test statistic (z):= ", z_stat)
print(f"One-sided p-value (P(Z ≥ z)) = ", p_value_one_sided)
print(f"Significance level (α): ", alpha, "Critical z (one-sided) ≈ ", z_crit_one_sided)


# Decision
if z_stat > z_crit_one_sided:
    print("Decision: Reject H0 (evidence that weekly costs are higher).")
else:
    print("Decision: Fail to reject H0 (no evidence that weekly costs are higher).")

Hypothesis Test: Are weekly costs higher than the theoretical model?
Sample mean (x̄):  3050.0
Theoretical mean (μ):  4000.0
Population sd of weekly cost:  125.0
Sample size (n):  25
Standard error :  25.0
Test statistic (z):=  -38.0
One-sided p-value (P(Z ≥ z)) =  1.0
Significance level (α):  0.05 Critical z (one-sided) ≈  1.6448536269514722
Decision: Fail to reject H0 (no evidence that weekly costs are higher).


## **CONCLUSION**

Z(-38.0) is not greater than critical value z = 1.64485.

So, **fail to reject H0.**

The sample mean 3050/- is lower than predicted theoretical mean 4000/-. So, the weekly operational costs are not higher than the theoretical models predicted.

**So, the test rejects the claim H0.**